packages/safe-web3-lib/src/Web3Adapter.ts

```diff
@@ -53,60 +48,53 @@ class Web3Adapter implements EthAdapter {
   getSafeContract({
     safeVersion,
     chainId,
-    isL1SafeMasterCopy,
+    singletonDeployment,
     customContractAddress,
     customContractAbi
-  }: GetSafeContractProps): GnosisSafeContractWeb3 {
-    const safeSingletonDeployment = getSafeContractDeployment(
-      safeVersion,
-      chainId,
-      isL1SafeMasterCopy
-    )
-    const contractAddress =
-      customContractAddress ?? safeSingletonDeployment?.networkAddresses[chainId]
+  }: GetContractProps): GnosisSafeContractWeb3 {
+    const contractAddress = customContractAddress ?? singletonDeployment?.networkAddresses[chainId]
     if (!contractAddress) {
       throw new Error('Invalid Safe Proxy contract address')
     }
     const safeContract = this.getContract(
       contractAddress,
-      customContractAbi ?? (safeSingletonDeployment?.abi as AbiItem[])
+      customContractAbi ?? (singletonDeployment?.abi as AbiItem[])
```

Comment:
Does this still need to be casted after the refactor? Or should it wait until [this](https://github.com/gnosis/safe-core-sdk/issues/138).

Reply:
The same goes for the other `AbiItem[]` cast below.

Reply:
`SingletonDeployment` type from `safe-deployments` library defines the `abi` as a `any[]`